src/BIP44CoinTypeNode.test.ts

```diff
@@ -356,6 +357,32 @@ describe('deriveBIP44AddressKey', () => {
     ).toStrictEqual(expectedKey);
   });
 
+  it('derives an address_index key (default inputs, hardened address_index)', () => {
+    const coinType = 60;
+    const parentNode = new BIP44CoinTypeNode([
+      defaultBip39NodeToken,
+      BIP44PurposeNodeToken,
+      `bip32:${coinType}'`,
+    ]);
+
+    const expectedKey = new BIP44Node({
+      derivationPath: [
+        defaultBip39NodeToken,
+        BIP44PurposeNodeToken,
+        `bip32:${coinType}'`,
+        `bip32:0'`,
+        `bip32:0`,
+        `bip32:3333'`,
+      ],
+    }).key;
+
+    expect(
+      deriveBIP44AddressKey(parentNode, {
+        address_index: BIP_32_HARDENED_OFFSET + 3333,
```

Comment:
Alternatively, instead of simply adding the hardened offset to the `address_index` (or `change`), we could also:

- Allow specifying a string, like `3333'`
- Allow specifying an object, like `{ index: number; isHardened: boolean; }`

Reply:
I would lean towards specifying an object. At least that matches the current API

Reply:
But perhaps we need an object for each index then? `deriveBIP44AddressKey(parentNode, { index: 60, isHardened: true }, (...))`

Reply:
Yeah, I was thinking of something like this:
```ts
deriveBIP44AddressKey(parentNode, {
  // Both `number` and `{ index: number; isHardened: boolean; }` supported
  change: 0,
  address_index: {
    index: 3333,
    isHardened: true
  }
});
```

Reply:
Yeah, that could also work 👍

Reply:
I like @Mrtenz's most recent object suggestion, with one tweak: let's rename the `isHardened` property to just `hardened`.

Reply:
Oh, I'm assuming that if you only specify a number, that will be used as the index. Since this is intended for BIP-44, we should default the hardening behavior to whatever BIP-44 specifies.